File: DI/DI-Dagger-Java/src/main/java/DI/persona/PersonasControllerFactory.java
package DI.persona;

import DI.persona.controllers.DaggerPersonasDBController;
import DI.persona.controllers.DaggerPersonasFileController;
import model.persona.controllers.PersonaController;

public class PersonasControllerFactory {
    private static PersonasControllerFactory instance = null;

    private PersonasControllerFactory() {
    }

    public static PersonasControllerFactory create() {
        if (instance == null) {
            instance = new PersonasControllerFactory();
        }
        return instance;
    }

    public PersonaController withDBStorage() {
        return DaggerPersonasDBController.create().build();
    }

    public PersonaController withFileStorage() {
        return DaggerPersonasFileController.create().build();
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/personas/PersonasControllerFactory.kt
package DI.personas

import DI.personas.controllers.DaggerPersonasDBController
import DI.personas.controllers.DaggerPersonasFileController
import models.personas.controllers.PersonaController


object PersonasControllerFactory {
    fun withDBStorage(): PersonaController {
        return DaggerPersonasDBController.create().build()
    }

    fun withFileStorage(): PersonaController {
        return DaggerPersonasFileController.create().build()
    }
}
